### PR TITLE
translate default calendar & default address book displaynames & add …

### PR DESCRIPTION
**Pull request by LukasReschke**
…color for calendar

From https://github.com/owncloud/core/pull/25887

### CODE REVIEW

**Comment by nickvergessen** [comment]
Fine by me, although I think that this should be done while outputing the data instead of storing it translated in the database. The advantage would be that when you change the language later on, it would translate to your new language then. Similar to https://github.com/nextcloud/server/pull/936

**Comment by rullzer** [comment]
I agree with @nickvergessen here. Lets store everything in english and translate on the fly when it is required.

**Comment by MorrisJobke** [comment]
> I agree with @nickvergessen here. Lets store everything in english and translate on the fly when it is required.

really? I would store the contacts address book localized and then the user can change it to whatever the user wants. Or did I miss something?

**Comment by nickvergessen** [comment]
But on user creation the language is english, so the calendar is english as well?

**Comment by MorrisJobke** [comment]
> But on user creation the language is english, so the calendar is english as well?

Ah right. It would be in the language of the admin. The weird thing then only is that when the user fetches the address book from a different machine with different locale the name of the address book changes. o.O or am I wrong here too?

**Comment by nickvergessen** [comment]
Same would be the case for activities, no. User language is before "sent language"

**Comment by MorrisJobke** [comment]
> User language is before "sent language"

Good :)

**Comment by rullzer** [comment]
So we can translate the display name of course. But if the user then explicitly sets it to 'Personal' we have no way of knowing that. So it will be translated again then.

File: apps/dav/lib/HookManager.php [hunk 1]
```diff
@@ -46,14 +47,19 @@ class HookManager {
...
 
+	/** @var L10N */
+	private $l10n;
+
 	public function __construct(IUserManager $userManager,
 								SyncService $syncService,
 								CalDavBackend $calDav,
-								CardDavBackend $cardDav) {
+								CardDavBackend $cardDav,
+								L10N $l10n) {
```
Comment (nickvergessen) [inline apps/dav/lib/HookManager.php]: use `OCP\L10N\IL10N` interface please

Comment (tcitworld) [inline apps/dav/lib/HookManager.php]: Done.

**Comment by nickvergessen** [comment]
Updated the PR, now it only translates if URI and name match the default.
So renaming is possible.

**Comment by tcitworld** [comment]
You'll need to add the dav app to transifex too for these to be translated, right ?

**Comment by nickvergessen** [comment]
It should already be, because we translate the birthday calendars already. But @MorrisJobke should check that just to be sure.

**Comment by tcitworld** [comment]
Well it's not in `https://github.com/nextcloud/server/blob/master/l10n/.tx/config`, for all I know.

**Comment by MorrisJobke** [comment]
> Well it's not in l10n/.tx/config, for all I know.

Correct. I will add it later.

**Comment by LukasReschke** [comment]
> Correct. I will add it later.

👋

**Comment by nickvergessen** [comment]
Rebased to solve the conflicts (Application magic DI PR)

**Comment by MorrisJobke** [comment]
> Correct. I will add it later.

added transifex config

File: l10n/.tx/config [hunk 2]
```diff
@@ -24,6 +24,12 @@ type = PO
 file_filter = <lang>/lib.po
 source_file = templates/lib.pot
 source_lang = en
+
+[nextcloud.dav]
+file_filter = <lang>/dav.po
+source_file = templates/dav.pot
+source_lang = en
+type = PO
 type = PO
```
Comment (nickvergessen) [inline l10n/.tx/config]: You had one job to do...

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
Rebased once more

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by rullzer** [comment]
LGTM!

**Pull request merged**